ce/api/percentileanomaly.py

```diff
@@ -183,62 +297,91 @@ def canonical_timestamp(timestamp, timescale, timeidx):
                                                             row['units']))
 
                     if row['climatology'] == climatology:
-                        add_to_nested_dict(projected_data,
-                                           row["mean"],
-                                           row["timescale"],
-                                           ctimestamp,
-                                           row["model"])
+
+                        projected_data, model_counter = add_to_nested_li(projected_data, row, ctimestamp, model_counter)
+
                     elif (calculate_anomaly and
                             row['model'] == baseline_model and
                             row['climatology'] == baseline_climatology):
-                        add_to_nested_dict(baseline_data, row['mean'],
-                                           row['timescale'], ctimestamp)
+
+                        baseline_data = add_to_nested_li(baseline_data, row, ctimestamp)
+
+
+
+        def determine_baseline(curr_idx, p_obj, baseline_data, baseline_model, baseline_climatology):
```

Comment:
Place this outside the loop, and for that matter, outside the `try`. It is (or at least should be) independent of any variables being maintained in the loop, and technically speaking you are redefining the function each time through the loop. You want to avoid that, mainly for clarity (the Python interpreter is pretty smart and will probably avoid actually recreating the function each time).